hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngine.java

```diff
@@ -16,122 +16,21 @@
 
 package com.hazelcast.map.impl.query;
 
-import com.hazelcast.query.PagingPredicate;
 import com.hazelcast.query.Predicate;
 import com.hazelcast.util.IterationType;
 
 import java.util.Set;
-import java.util.concurrent.ExecutionException;
 
 /**
  * Responsible for executing queries on the IMap.
  */
 public interface MapQueryEngine {
 
-    /**
-     * Executes a query on all the local partitions.
-     * <p>
-     * - Uses Indexes
-     * - Accepts PagingPredicate
-     * - Query executed in the calling thread
-     * - predicate evaluation will be parallelized if QUERY_PREDICATE_PARALLEL_EVALUATION is enabled or a PagingPredicate is used.
-     * - may return empty QueryResult (with a properly initialized result limit size) if query is executed during migrations and
-     * migration conditions do not allow for safe retrieval of results (ie. there are migrations in flight or partition state
-     * version is different at the end of query vs the version observed at the beginning of query). In this case, partitionIds
-     * are not set on the QueryResult, so callers will ignore these results and fallback to {@code QueryPartitionOperation}s.
-     *
-     * @param mapName   the name of the map
-     * @param predicate the predicate
-     * @return the QueryResult
-     * @throws ExecutionException
-     * @throws InterruptedException
-     */
-    QueryResult queryLocalPartitions(String mapName, Predicate predicate, IterationType iterationType)
-            throws ExecutionException, InterruptedException;
+    Set runQueryOnAllPartitions(String mapName, Predicate predicate, IterationType iterationType, boolean uniqueResult);
```

Comment:
Would it not be easier if:
1: introduce a query object which contains what you actually want to do. Like single partition, multiple partitions etc. Also when new parameters are added; with a query the interface won't break that quick
2: make it async. This way we don't block the generic thread.